compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2347,6 +2350,60 @@ public void visit(BLangTupleDestructure tupleDeStmt, AnalyzerData data) {
         if (type.tag != TypeTags.SEMANTIC_ERROR) {
             checkTupleVarRefEquivalency(tupleDeStmt.pos, tupleDeStmt.varRef,
                                         tupleDeStmt.expr.getBType(), tupleDeStmt.expr.pos, data);
+            if (tupleDeStmt.expr.getKind() == LIST_CONSTRUCTOR_EXPR) {
+                analyzeDestructuringWithListBindingPattern(tupleDeStmt.varRef,
+                        (BLangListConstructorExpr) tupleDeStmt.expr);
+            }
+        }
+    }
+
+    private void analyzeDestructuringWithListBindingPattern(BLangTupleVarRef tupleVarRef,
+                                                            BLangListConstructorExpr listExpr) {
+        List<BLangExpression> exprs = listExpr.exprs;
+        List<BLangExpression> tupleVars = tupleVarRef.expressions;
+
+        int nonRestTupleVarIndex = 0;
+        int noOfNonRestVars = tupleVars.size();
+
+        for (BLangExpression expr: exprs) {
+            int remainingNonRestVars = noOfNonRestVars - nonRestTupleVarIndex;
+
+            if (remainingNonRestVars == 0) {
+                return;
+            }
+
+            if (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_SPREAD_OP) {
+                BType spreadOpType = ((BLangListConstructorSpreadOpExpr) expr).expr.getBType();
+                int noOfSpreadMembers = switch (spreadOpType.tag) {
+                    case TypeTags.ARRAY -> {
+                        BArrayType spreadArrayType = (BArrayType) spreadOpType;
+                        yield spreadArrayType.size;
+                    }
+                    case TypeTags.TUPLE -> {
+                        BTupleType spreadTupleType = (BTupleType) spreadOpType;
+                        yield spreadTupleType.getTupleTypes().size();
+                    }
+                    default -> 0;
+                };
+
+                if (remainingNonRestVars < noOfSpreadMembers) {
+                    return;
+                }
+
+                nonRestTupleVarIndex += noOfSpreadMembers;
+                continue;
+            }
+
+            BLangExpression tupleMemberVar = tupleVars.get(nonRestTupleVarIndex++);
+            if (expr.getKind() == LIST_CONSTRUCTOR_EXPR && tupleMemberVar.getKind() == TUPLE_VARIABLE_REF) {
```

Comment:
Codecov warning?

Reply:
The following test covers this.

```
[[e7, e8]] = [[...[...[employee1, employee2]]]];
```

already added to https://github.com/ballerina-platform/ballerina-lang/pull/40283/files#diff-134c57a0d31d5ced27fd93ca77e8b0c17998533bcd4e91666477901565e5ead3